scrapy/core/engine.py

```diff
@@ -250,11 +250,10 @@ def _remove_request(_: Any) -> None:
         )
         return d2
 
+    @inlineCallbacks
     def _handle_downloader_output(
```

Comment:
Did you consider `async def` with the deferred decorator? I know we cannot use it at some key places, but I wonder if this is one of them.